### ISSUE TITLE
Suggestion for improvement of the Link and UriTemplate classes

### ISSUE THREAD

**Issue by wis3guy** [JakeGinnivan/WebApi.Hal#46]
Currently, when you specify a link as a template like this:

```
~/users/{userid}/friends/{friendid}/interactions
```

... and then, in the UserRepresentation, where i know the value for the {userid} but _not_ for the {friendid}, i want to return the above link, i need to call:

```
Links.Add(LinkTemplates.Users.InteractionList.CreateLink(new {userid = 1}));
```

As a result my url will become: /users/1/friends//interactions

What i would like in this case, is that the Link class, and the backing UriTemplate class would return a new Link instance, which is still a template, having the following href: ~/users/1/friends/{friendid}/interactions

In my use-case, the value for {friendid}, is known only to the consumer. I do not have these values in my backing data store. I actually would like to return a templated link to my clients.

For now, i work around the issue by defining this link like this:

```
~/users/{userid}/interactions{?friendid}
```

In conclusion, I suggest that the CreateLink, CreateUri and SubstituteParams methods get overloads that will result in partially resolved uri's and thus links that will be serialized in the response with the "templated" (http://tools.ietf.org/html/draft-kelly-json-hal-06#section-5.1) value set accordingly.

**Comment by panmanphil** [comment]
closing stale issue
